sonarqube/assets/configuration/spec.yaml

```diff
@@ -58,6 +58,8 @@ files:
           items:
             type: string
       - name: components
+        required: true
```

Comment:
This should continue being not required. It was initilised as an empty dict if so

`parse_config`
```
components = self.instance.get('components', {})
```

Reply:
However, the integration fails if it's not set so it's technically not optional.

```yaml
    ## @param components - mapping - optional
```

```
        elif not components:
            raise ConfigurationError('The `components` setting must be defined')
```